wear/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java

```diff
@@ -0,0 +1,172 @@
+    private static volatile int s_acumulatedSize = 0;
+
+    public static boolean isAtom() {
+        final ActiveBluetoothDevice activeBluetoothDevice = ActiveBluetoothDevice.first();
+        if (activeBluetoothDevice == null || activeBluetoothDevice.name == null) {
+            return false;
+        }
+        return activeBluetoothDevice.name.contentEquals("Atom");
+    }
+
+
+    public static BridgeResponse getAtomResponse() {
+        final BridgeResponse reply = new BridgeResponse();
+        ByteBuffer ackMessage = ByteBuffer.allocate(6);
+        ackMessage.put(0, (byte) 0x02);
+        ackMessage.put(1, (byte) 0x01);
+        ackMessage.put(2, (byte) 0x00);
+        ackMessage.put(3, (byte) 0x00);
+        ackMessage.put(4, (byte) 0x00);
+        ackMessage.put(5, (byte) 0x2B);
+        reply.add(ackMessage);
+        return reply;
+    }
+
+    public static int lens = 344;
+    public static int ATOM_FOOTER = 8;
+
+    static int errorCount = 0;
+
+
+    static byte[] patchUid = null;
+    static byte[] patchInfo = null;
+
+    public static BridgeResponse decodeAtomPacket(byte[] buffer, int len) {
+        final BridgeResponse reply = new BridgeResponse();
+        int first = 0xff & buffer[0];
+        if (first == 0x80) {
+            PersistentStore.setString("Atombattery", Integer.toString(buffer[4]));
+            Pref.setInt("bridge_battery", buffer[4]);
+            String atomfirmware = buffer[2] + "." + buffer[3];
+            String atomHArdware = buffer[buffer.length-2] + "." + buffer[buffer.length-1];
+            PersistentStore.setString("AtomHArdware", atomHArdware);
+            PersistentStore.setString("AtomFirmware", atomfirmware);
+            ByteBuffer ackMessage = ByteBuffer.allocate(6);
+            ackMessage.put(0, (byte) 0x02);
+            ackMessage.put(1, (byte) 0x01);
+            ackMessage.put(2, (byte) 0x00);
+            ackMessage.put(3, (byte) 0x00);
+            ackMessage.put(4, (byte) 0x00);
+            ackMessage.put(5, (byte) 0x2B);
+            reply.add(ackMessage);
+            s_full_data = null;
+            return getAtomResponse();
+        }
+        if (first == 0xC0) {
+            patchUid = Arrays.copyOfRange(buffer, 2, 10);
+            String SensorSn = LibreUtils.decodeSerialNumberKey(patchUid);
+            PersistentStore.setString("LibreSN", SensorSn);
+            return reply;
+        }
+        if (first == 0xC1) {
+            double fv = JoH.tolerantParseDouble(PersistentStore.getString("AtomFirmware"));
+            if (fv < 1.35) {
+                patchInfo = Arrays.copyOfRange(buffer, 3, 9);
+            } else {
+                if (buffer.length >= 11) {
+                    patchInfo = Arrays.copyOfRange(buffer, 5, 11);
+                }
+            }
+            return reply;
+        }
+        if (first == 0x82) {
+            int expectedSize = lens + ATOM_FOOTER;
+            if (s_full_data == null) {
+                InitBuffer(expectedSize)
```

Comment:
missing `;` here by the way

Reply:
Thank you for your comment and I ask you to excuse me for my mistake. Fixed